src/Neo/SmartContract/Native/GasToken.cs

```diff
@@ -43,6 +43,14 @@ internal override async ContractTask OnPersistAsync(ApplicationEngine engine)
             {
                 await Burn(engine, tx.Sender, tx.SystemFee + tx.NetworkFee);
                 totalNetworkFee += tx.NetworkFee;
+
+                // Reward for NotaryAssisted attribute will be minted to designated notary nodes
+                // by Notary contract.
+                var notaryAssisted = tx.GetAttribute<NotaryAssisted>();
+                if (notaryAssisted is not null)
+                {
+                    totalNetworkFee -= (notaryAssisted.NKeys + 1) * Policy.GetAttributeFee(engine.SnapshotCache, (byte)notaryAssisted.Type);
```

Comment:
@AnnaShaleva `AttributeFee` in memory pool can be 1, and during the block persistence can be changed to 2, this could underflow `totalNetworkFee` and produce a denial of service.

Reply:
Can't happen. It's OnPersist and it's executed before any in-block transaction, the value is the same as was used when verifying the transaction.

Reply:
@shargon, do you still have any concern about this issue?

Reply:
I'm agree with @roman-khimov, it will be the same as other attributes, and other fees, but i'm not sure if this is the best way to recollect fees, why not deposit the Gas in the contract as a fee during the execution of the contract (like a regular contract can do)?